lib/BackgroundJob/RemoveEmptyRooms.php

```diff
@@ -47,15 +47,23 @@ class RemoveEmptyRooms extends TimedJob {
 	/** @var LoggerInterface */
 	protected $logger;
 
-	/** @var FederationManager */
-	protected $federationManager;
+	/** @var IUserMountCache */
+	protected $userMountCache;
 
 	protected $numDeletedRooms = 0;
 
+	/**
+	 * The room to proccess
+	 *
+	 * @var Room|null
+	 */
+	protected $room = null;
```

Comment:
Made some bad experiences lately as cron jobs keep the state across multiple executions within the same `cron.php` call. It's not a problem here, but I would avoid the pattern. Can we instead pass around the room object?

Reply:
The room object is changed on all executions. We can pass the object between the methods but will make the code biggest.

Reply:
Do you want that I thing on another approach?

Reply:
You could just pass in $room into the other methods?